rust/otel-arrow-rust/src/encode/record/array/string.rs

```diff
@@ -0,0 +1,179 @@
+use std::sync::Arc;
+
+use arrow::array::{
+    Array, ArrowPrimitiveType, DictionaryArray, StringArray, StringBuilder, StringDictionaryBuilder,
+};
+use arrow::datatypes::{ArrowDictionaryKeyType, DataType, UInt8Type, UInt16Type};
+use arrow::error::ArrowError;
+
+use super::dictionary::{DictionaryArrayBuilder, DictionaryBuilderError as Error, Result};
+use super::{ArrayBuilder, ArrayBuilderConstructor, dictionary::UpdateDictionaryIndexInto};
+
+impl ArrayBuilderConstructor for StringBuilder {
+    fn new() -> Self {
+        Self::new()
+    }
+}
+
+impl ArrayBuilder for StringBuilder {
+    type Native = String;
+
+    fn append_value(&mut self, value: &Self::Native) {
+        self.append_value(value);
+    }
+
+    fn finish(&mut self) -> super::ArrayWithType {
+        return super::ArrayWithType {
+            data_type: DataType::Utf8,
+            array: Arc::new(self.finish()),
+        };
+    }
+}
+
+impl<T> ArrayBuilderConstructor for StringDictionaryBuilder<T>
+where
+    T: ArrowDictionaryKeyType,
+{
+    fn new() -> Self {
+        Self::new()
+    }
+}
+
+impl<T> DictionaryArrayBuilder for StringDictionaryBuilder<T>
+where
+    T: ArrowDictionaryKeyType + ArrowPrimitiveType,
+    <T as ArrowPrimitiveType>::Native: Into<usize>,
+{
+    type Native = String;
+
+    fn append_value(&mut self, value: &Self::Native) -> Result<usize> {
+        match self.append(value) {
+            Ok(index) => Ok(index.into()),
+            Err(ArrowError::DictionaryKeyOverflowError) => Err(Error::DictOverflow {}),
+
+            // safety: shouldn't happen. The only error type append should
+            // return should be for dictionary overflows
+            Err(e) => panic!("{}", e),
```

Comment:
If that happen we probably need more context.

Reply:
Made this change to give more context